typegraph/core/src/types/type_def.rs

```diff
@@ -13,6 +14,7 @@ use crate::wit::core::{
 };
 use common::typegraph::{Injection, TypeNode};
 use enum_dispatch::enum_dispatch;
+use std::hash::Hash as _;
```

Comment:
We have two places that requires some form of caching right.. (typegate introspection + SDK) I think investigating on the typegate side might also give more insights at what factor we can realistically reduce the type size before implemenenting it, I mean.. if the most effective reduction scheme is still slower than the actual growth as there are a lot of unique types we might have to think of alternative solutions 🤔.. also I am curious, have you already tried this approach on a heavy typegraph?